server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -252,8 +252,14 @@ public void run()
           try {
             try {
               inputRow = firehose.nextRow();
+
+              if (inputRow == null) {
+                log.debug("thrown away null input row, considering unparseable");
```

Comment:
That also seems like it would add a lot of overhead for what is arguably a corner case.

Ultimately it would be nice if "where did my failure happen" had some sort of breadcrumb here, but given the nature of ingestion it probably makes more sense to have a validation layer well before this point in the data stream (aka not in druid).  And any failures here could be assumed to be fault in the communication, making this a safeguard to prevent catastrophic behavior on rare events.

Does that sound like a correct statement?